### ISSUE TITLE
question: public API?

### ISSUE THREAD

**Issue by mizlan** [hotseatio/hotseat.io#92]
Is there a public API for Hotseat? I specifically would be interested in the class enrollment data. I began work on scraping it myself, mostly successfully, but it seems that would be duplicating efforts that have already been done.

**Comment by nathanmsmith** [comment]
There is not a public API for Hotseat, but that is something we've been talking about making. @mizlan, what's your planned usage with the enrollment data and would you be interested in contributing to the public API?

**Comment by mizlan** [comment]
I want to design my own frontend for the calendar-view planner, and I am
more generally interested in the decoupling of the enrollment data from any
particular interface.

I would be, yes! I also would like to ask: would you be opposed to offering
some method of exporting a dump of the current state of the enrollment data?

**Comment by nathanmsmith** [comment]
> would you be opposed to offering some method of exporting a dump of the current state of the enrollment data?

Wouldn't the API do this? I think an API would help smooth over some of the rough edges in our database schema too, so I think I'd prefer that to a dump. Let me know if I'm missing something though.

**Comment by nathanmsmith** [comment]
fwiw, we've also been thinking about making a Hotseat-ified version of the calendar-view planner – I'm curious about what improvements you're thinking of! Mind if we set up a call? I'd love to learn more and get your thoughts on the API design. I can help get your env set up for Hotseat too!

**Comment by mizlan** [comment]
> Wouldn't the API do this

Yeah, but not the converse. It would be simpler for now, I guess (?) But you're right that an API would be helpful in locating any rough edges in the database.

> fwiw, we've also been thinking about making a Hotseat-ified version of the calendar-view planner – I'm curious about what improvements you're thinking of! Mind if we set up a call? I'd love to learn more and get your thoughts on the API design. I can help get your env set up for Hotseat too!

Oooh, that sounds good! Is hi@nathansmith.io a good address to reach you at?

**Comment by nathanmsmith** [comment]
@mizlan yes it is!
